File: XMLCheck/Handler/CheckRulesXML11/Xml11RuleBase.cs
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML11
{
    public abstract class Xml11RuleBase:IXml11Rule
    {
        public abstract string Key { get; }

        public abstract ErrorDetails? Check(Xml11Model model,HoSoContext context);

        protected ErrorDetails Error(string message)
        {
            return new ErrorDetails
            {
                key = Key,
                message = message
            };
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml11/xml11RuleBase.ts
import { Xml11Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';
import { BaseRule } from '../base.rule';

export interface IXml11Rule {
  readonly key: string;
  readonly isEnabled: boolean;
  check(model: Xml11Model, context: HoSoContext): ErrorDetails | null;
}

export abstract class Xml11RuleBase extends BaseRule<Xml11Model> implements IXml11Rule {}
